File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int result = nums[0], count = 1;
        for (int i = 1; i < nums.length; i++) {
            if (nums[i] == result) {
                count++;
            }
            else {
                count--;
                if (count == 0) {
                    result = nums[i];
                    count = 1;
                }
            }
        }
        return result;
    }
}

File: 0169-majority-element/0169-majority-element.kt
class Solution {
    fun majorityElement(nums: IntArray): Int {
        var result = nums[0]
        var count=0
        for(i in nums){
            if(i==result){
                count++
            }else{
                count--
                if(count==0){
                    result = i
                    count =1
                }
                
            }
        }
        return result
    }
}